src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt

```diff
@@ -105,6 +105,25 @@ object ValidationUtil {
             validationException.addValidationError("Value $name must not start with '.'")
     }
 
+    /**
+     * Validate the pattern against the rules that we have for indexPattern name.
+     */
+
+    fun validatePattern(pattern: String?, validationException: ValidationException) {
+
+        if (!Strings.validFileNameExcludingAstrix(pattern))
```

Comment:
The characters in the pattern doesn't have to a valid filename right? Wanted to understand the how you arrived at this criteria.

Reply:
I have tried to adapt similar validations done in opensearch on indexPatterns at index template level where the index Pattern must not contain invalid filename characters. 

reference: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/metadata/MetadataIndexTemplateService.java